bundler/lib/bundler/plugin/index.rb

```diff
@@ -111,6 +111,17 @@ def command_plugin(command)
         @commands[command]
       end
 
+      # Plugin cannot be installed and updating to install
+      def cannot_install?(name, version)
+        installed?(name) && !updating?(name, version)
```

Comment:
The `cannot_install?` makes me think of same problem when trying to install the gem. Maybe `version_already_installed?` or something like that would be better?

Reply:
Really sounds better, I changed the def name. Also want ask about the plugin version, do you have any suggest to how get it without using the regex, maybe a build-in way that I missed... or this way is fine? I'm talking about plugin index [here](https://github.com/dfop02/rubygems/blob/fix-bundle-plugin-bug/bundler/lib/bundler/plugin/index.rb#L121).

Reply:
I had the same thought but I could not find a better way that the regex based approach you used.

Reply:
My initial objective was separate the methods to looks simple, because already exists the `installed?` so I just added the `!updating?`, maybe change to `not_updating?` or `same_version?` feels more readable. But since still sounds confuse, I can try your approach. Also, good idea about the "regex issue", I'll test :)

Reply:
While I was testing your approach about the "regex issue", I found few more things to fix, and now looks like *almost everything works fine. I notice that the index file was not updating the path correctly after change versions updating/downgrading so I needed fix it too.

Almost because before, the plugin was suppose to appears "Using foo x.y.z" when already installed, but now is suppressed by default (only for plugins), I'm not sure if is a big problem. That happens because when you cli bundle install, the [installed_specs](https://github.com/dfop02/rubygems/blob/fix-bundle-plugin-bug/bundler/lib/bundler/source/rubygems.rb#L451-L453) returns the plugin while bundle by gemfile doesnt and I couldnt understand why that happens.

Reply:
Thanks for all the work, I'll loop back and have a look at that last issue.

Reply:
@dfop02 Sorry I didn't loop back to this yet. Did you look into it? We recently removed unnecessary "Using..." messages at #6804, not sure if that changes anything here.

Reply:
Hello @deivid-rodriguez !

I take a look again today on currently bundler version, looks like the updates solve the "using..." messages that I told before, what is nice, but installing messages and index file issue continues. For now I create a separate matcher because I thought that was causing the failures on specs, let's see if solves it :)